include/zephyr/dt-bindings/clock/stm32u5a9_clock.h

```diff
@@ -0,0 +1,21 @@
+/*
```

Comment:
Do we really need this file? Shouldn't we include the new clock sources in the stm32u5_clock.h file directly?
I don't see any reasons to make it separate, and if we want to keep things consistent, we should do this kind of thing across all series, and we'd end up with far too many files.
cc @erwango

Reply:
These are peripherals than are available only on specific U5 SoCs, namely STM32U59x/5Ax & STM32U5Fx/5Gx, you can check "Memory organization->Memory map and register boundary addresses -> Table 6" in the following doc for more details.
https://www.st.com/resource/en/reference_manual/rm0456-stm32u5-series-armbased-32bit-mcus-stmicroelectronics.pdf

Reply:
I know that these peripherals are not available on all U5, but that happens on all series. As we go up in the nomenclature, more peripherals are available. If we start making specific include file every time a higher-end SoC includes a new peripheral, we'll fill the folder with our clock files.
Currently, we have one clock file per series (except for F4), with some defines that are not applicable to some SoC, but that isn't a problem. Describing the available peripherals is the role of the dtsi, not these includes. So I think we can stick with only one per series.